### DISCUSSION TITLE
[PSD3] Esercizio Slide 3

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#135] [Esercizi (A-L)]
Convertire i seguenti numeri in IEEE 754 half-precision e scrivere la codifica esadecimale corrispondente

(a) $−1.5625 × 10^{−1}$
(b) $1.771 × 10^3$
(c) $0.34375$

**Comment by Elia-Belli** [comment]
**Half-precision (16bit: 1-5-10)**
(a) $-1.5625 * 10^{-1} = -0.15625 = -0.00101 * 2^0 = 1.01* 2^{-3} \rightarrow 1\ 01100\ 0100000000 = B100_{16} $
(b) $1.771 * 10^3 = 1771 = 11011101011* 2^0 = 1.1011101011 * 2^{10} \rightarrow 0\ 11001\ 1011101011 = 66EB_{16}$
(c) $0.34375 = 0.01011 * 2^0 = 1.011 * 2^{-2} \rightarrow 0\ 01101\ 0110000000 = 3580_{16}$

**Comment by FeddyLix17** [comment]
(a) corretto
(b) corretto tranne la base 16 (dovrebbe essere 66EB perchè 1110 fa 14 che è E)
(c) corretto

**Comment by Elia-Belli** [comment]
Sistemato, grazie
